test/support/cli_helpers.ex

```diff
@@ -9,26 +9,46 @@ defmodule Support.CliHelpers do
   def run("cogctl" <> args) do
     capture_io(fn ->
       capture_io(:stderr, fn ->
-        try do
-          args
-          |> String.split
-          |> smart_split([])
-          |> Cogctl.main
-        catch
-          _, _ ->
-            nil
-        end
+        try_run("cogctl" <> args)
       end) |> IO.write
     end)
   end
   def run(_) do
     raise ~s(Commands must start with "cogctl")
   end
 
+  def run_capture_stderr("cogctl" <> args) do
+    capture_io(:stderr, fn ->
+      capture_io(fn ->
+        try_run("cogctl" <> args)
+      end)
+    end)
+  end
+
+  def run_capture_stdio("cogctl" <> args) do
+    capture_io(fn ->
+      capture_io(:stderr, fn ->
+        try_run("cogctl" <> args)
+      end)
+    end)
+  end
```

Comment:
I agree it is a bit odd, but I'm not sure another way to do it. If I just do `capture_io(:stderr, fn)` then stderr is captured but stdout is printed. If I do capture_io(fn)` then stdout is captured but stderr is printed. If I use `run` then both are captured, but the order is inconsistent and it's difficult to match on in the tests. This way I can match on either output and not have the other print during the tests.